File: 119-pascals-triangle-ii/pascals-triangle-ii.js
function getRow(rowIndex) {
    const row = [1]; // First element is always 1

    for (let i = 1; i <= rowIndex; i++) {
        // Compute next element using previous one
        const nextVal = row[i - 1] * (rowIndex - i + 1) / i;
        row.push(nextVal);
    }

    return row;
}

// Examples
console.log(getRow(3)); // Output: [1, 3, 3, 1]
console.log(getRow(0)); // Output: [1]
console.log(getRow(1)); // Output: [1, 1]

File: 119-pascals-triangle-ii/pascals-triangle-ii.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> row;
        row.push_back(1); // First element is always 1

        long long val = 1; // Use long long to avoid overflow
        for (int i = 1; i <= rowIndex; i++) {
            val = val * (rowIndex - i + 1) / i;
            row.push_back((int)val); // Convert back to int
        }

        return row;
    }
};
